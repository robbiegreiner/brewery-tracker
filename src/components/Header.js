import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import pint from '../assets/pint.svg';


const Header = () => {
  function dropDown() {
    document.getElementById("myDropdown").classList.toggle("show");
  } 

  window.onclick = function(event) {
    if (!event.target.matches('.dropbtn')) {

      var dropdowns = document.getElementsByClassName("dropdown-content");
      var i;
      for (i = 0; i < dropdowns.length; i++) {
        var openDropdown = dropdowns[i];
        if (openDropdown.classList.contains('show')) {
          openDropdown.classList.remove('show');
        }
      }
    }
  };



  return (
    <div className='header'>
      <Link to='/'>
        <h1>Brewery <img className='header-logo' src={pint}></img> Tracker</h1>
      </Link>
      <div className="dropdown">
        <button onClick={dropDown} className="dropbtn">Dropdown</button>
        <div id="myDropdown" className="dropdown-content">
          <a href="#">Link 1</a>
          <a href="#">Link 2</a>
          <a href="#">Link 3</a>
        </div>
      </div>
    </div>
  );
};

export default Header;
