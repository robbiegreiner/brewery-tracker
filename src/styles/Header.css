.header {
  background: #2C4154;
  width: 100%;
  height: 10vh;
  min-height: 94px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 0px;
  color: #FF644E;
  position: fixed;
  border-bottom: 3px solid #FF644E;
  font-family: 'Rubik', sans-serif;
  font-size: 20px;
  font-style: italic;
  font-weight: 900;
  z-index: 1; }

.header img {
  height: 50px;
  width: 50px; }

.header h3 {
  cursor: pointer; }

.header a {
  color: white; }

.header h1 {
  text-align: center; }

.header-buttons {
  position: absolute;
  right: 0px;
  display: flex; }

.header-buttons h3 {
  font-style: normal;
  font-weight: 500;
  margin-right: 20px;
  padding: 0;
  text-align: right; }

.logged-in-buttons {
  font-style: normal;
  font-weight: 500;
  position: absolute;
  right: 10px;
  bottom: 50px; }

.logged-in-buttons h5 {
  font-size: 12px;
  font-weight: 400; }

.logout {
  font-size: 16px;
  font-weight: 400; }

.logged-in-buttons .fav-button {
  position: absolute;
  top: 50px;
  right: 20px; }

@media screen and (max-width: 1160px) {
  .header {
    justify-content: start; }
  .header h1 {
    margin-left: 40px; } }
