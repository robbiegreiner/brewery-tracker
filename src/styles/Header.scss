.header {
  background-color: #eb9122;
  background-image: url("https://www.transparenttextures.com/patterns/fake-brick.png");
  width: 100%;
  height: 10vh;
  min-height: 94px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 0px;
  color: white;
  position: fixed;
  border-bottom: 3px solid white;
}

.header img {
  height: 70px;
  width: 70px;
}

a {
  color: white;
}
