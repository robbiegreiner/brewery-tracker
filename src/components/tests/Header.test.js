import React from "react";
import Header from '../Header';
import { shallow } from "enzyme";

describe('Header component', () => {
  let wrapper;


  it('should match snapshot', () => {
    wrapper = shallow(<Header />)
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });


});
