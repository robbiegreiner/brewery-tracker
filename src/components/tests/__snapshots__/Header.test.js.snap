// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/login"
>
          <h3>
                    Login
          </h3>
</Link>,
        <Link
          replace={false}
          to="/createaccount"
>
          <h3>
                    Create Account
          </h3>
</Link>,
        <h3
          className="logout"
          onClick={undefined}
>
          Logout
</h3>,
        <Link
          replace={false}
          to="/"
>
          <h1>
                    Brew 
                    <img
                              className="header-logo"
                              src="pint.svg"
                    />
                     Scout
          </h1>
</Link>,
        <Link
          replace={false}
          to="/favorites"
>
          <h3>
                    Favorites
          </h3>
</Link>,
        <div
          className="dropdown"
>
          <button
                    className="dropbtn"
                    onClick={[Function]}
          >
                    Dropdown
          </button>
          <div
                    className="dropdown-content"
                    id="myDropdown"
          >
                    <a
                              href="#"
                    >
                              Link 1
                    </a>
                    <a
                              href="#"
                    >
                              Link 2
                    </a>
                    <a
                              href="#"
                    >
                              Link 3
                    </a>
          </div>
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h3>
            Login
</h3>,
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h3>
            Create Account
</h3>,
          "replace": false,
          "to": "/createaccount",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "className": "logout",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Logout",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            Brew 
            <img
                        className="header-logo"
                        src="pint.svg"
            />
             Scout
</h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Brew ",
              <img
                className="header-logo"
                src="pint.svg"
/>,
              " Scout",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Brew ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "className": "header-logo",
                "src": "pint.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            " Scout",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h3>
            Favorites
</h3>,
          "replace": false,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="dropbtn"
              onClick={[Function]}
>
              Dropdown
</button>,
            <div
              className="dropdown-content"
              id="myDropdown"
>
              <a
                            href="#"
              >
                            Link 1
              </a>
              <a
                            href="#"
              >
                            Link 2
              </a>
              <a
                            href="#"
              >
                            Link 3
              </a>
</div>,
          ],
          "className": "dropdown",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Dropdown",
              "className": "dropbtn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Dropdown",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="#"
>
                  Link 1
</a>,
                <a
                  href="#"
>
                  Link 2
</a>,
                <a
                  href="#"
>
                  Link 3
</a>,
              ],
              "className": "dropdown-content",
              "id": "myDropdown",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Link 1",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Link 1",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Link 2",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Link 2",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Link 3",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Link 3",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/login"
>
            <h3>
                        Login
            </h3>
</Link>,
          <Link
            replace={false}
            to="/createaccount"
>
            <h3>
                        Create Account
            </h3>
</Link>,
          <h3
            className="logout"
            onClick={undefined}
>
            Logout
</h3>,
          <Link
            replace={false}
            to="/"
>
            <h1>
                        Brew 
                        <img
                                    className="header-logo"
                                    src="pint.svg"
                        />
                         Scout
            </h1>
</Link>,
          <Link
            replace={false}
            to="/favorites"
>
            <h3>
                        Favorites
            </h3>
</Link>,
          <div
            className="dropdown"
>
            <button
                        className="dropbtn"
                        onClick={[Function]}
            >
                        Dropdown
            </button>
            <div
                        className="dropdown-content"
                        id="myDropdown"
            >
                        <a
                                    href="#"
                        >
                                    Link 1
                        </a>
                        <a
                                    href="#"
                        >
                                    Link 2
                        </a>
                        <a
                                    href="#"
                        >
                                    Link 3
                        </a>
            </div>
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h3>
              Login
</h3>,
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h3>
              Create Account
</h3>,
            "replace": false,
            "to": "/createaccount",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Create Account",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "className": "logout",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Logout",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              Brew 
              <img
                            className="header-logo"
                            src="pint.svg"
              />
               Scout
</h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Brew ",
                <img
                  className="header-logo"
                  src="pint.svg"
/>,
                " Scout",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Brew ",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "header-logo",
                  "src": "pint.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              " Scout",
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h3>
              Favorites
</h3>,
            "replace": false,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="dropbtn"
                onClick={[Function]}
>
                Dropdown
</button>,
              <div
                className="dropdown-content"
                id="myDropdown"
>
                <a
                                href="#"
                >
                                Link 1
                </a>
                <a
                                href="#"
                >
                                Link 2
                </a>
                <a
                                href="#"
                >
                                Link 3
                </a>
</div>,
            ],
            "className": "dropdown",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Dropdown",
                "className": "dropbtn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Dropdown",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href="#"
>
                    Link 1
</a>,
                  <a
                    href="#"
>
                    Link 2
</a>,
                  <a
                    href="#"
>
                    Link 3
</a>,
                ],
                "className": "dropdown-content",
                "id": "myDropdown",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Link 1",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Link 1",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Link 2",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Link 2",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Link 3",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Link 3",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
